pkg/task/cluster.go

```diff
@@ -20,9 +18,9 @@ func (ti *TaskInterface) ProcessClusters(directory string) error {
 		return err
 	}
 
-	cluster := retrieveClusterFromEnvironment()
+	cluster := &model.ClusterDetails{}
 
-	if cluster != nil && spec.Cluster != nil {
+	if spec.Cluster != nil {
```

Comment:
You could probably simplify this to not use `mergo`, which is pretty heavyweight. I.e.:

```
cluster := spec.Cluster
if cluster == nil {
  cluster = &model.ClusterDetails{}
}
```